File: src/main/java/com/moira/itda/domain/login/service/LogoutService.java
package com.moira.itda.domain.login.service;

import com.moira.itda.domain.login.mapper.LoginMapper;
import com.moira.itda.global.auth.CookieHandler;
import com.moira.itda.global.auth.SimpleUserAuth;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class LogoutService {
    private final CookieHandler cookieHandler;
    private final LoginMapper loginMapper;

    /**
     * 로그아웃
     */
    @Transactional
    public void logout(SimpleUserAuth userAuth, HttpServletResponse httpServletResponse) {
        // RTK 삭제 (DB, 쿠키)
        loginMapper.updateUserWhenLogout(userAuth.userId());
        cookieHandler.removeRtkInCookie(httpServletResponse);
    }
}

File: src/main/kotlin/com/moira/itda/domain/user_temp/logout/service/LogoutService.kt
package com.moira.itda.domain.user_temp.logout.service

import com.moira.itda.domain.user_temp.logout.mapper.LogoutMapper
import com.moira.itda.global.auth.component.CookieHandler
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class LogoutService(
    private val cookieHandler: CookieHandler,
    private val logoutMapper: LogoutMapper
) {
    /**
     * 상단바 > 로그아웃
     */
    @Transactional
    fun logout(userId: String, httpServletResponse: HttpServletResponse) {
        // [1] RTK 제거 (DB)
        logoutMapper.updateRefreshTokenNull(userId = userId)

        // [2] RTK 제거 (쿠키)
        cookieHandler.removeRtkInCookie(response = httpServletResponse)
    }
}